LeaderboardBackend/Services/Impl/RunService.cs

```diff
@@ -58,6 +59,54 @@ Page page
         return new ListResult<Run>(items, count);
     }
 
+    public async Task<GetRecordsForCategoryResult> GetRecordsForCategory(
+        long id,
+        Page page
+    )
+    {
+        Category? cat = await applicationContext.FindAsync<Category>(id);
+
+        if (cat is null)
+        {
+            return new NotFound();
+        }
+
+        IQueryable<Run> initQuery = applicationContext.Runs.FromSql($"""
+        SELECT *
+        FROM (
+            SELECT r.id, r.category_id, r.created_at, r.deleted_at, r.info, r.played_on, r.time_or_score, r.updated_at, r.user_id, RANK() OVER (PARTITION BY r.user_id ORDER BY r.time_or_score, r.played_on, r.created_at, r.id) as rank
```

Comment:
This needs to respect the category sort direction. You can use `FromSqlRaw()` for this purpose. Also, the inner SELECT can be simplified by using a star instead of listing all of the run table's columns.